### PR TITLE
fix(console): should not render when `watch` return `undefined` in sign up and sign in form

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

fix(console): should not render when `watch` return `undefined` in sign up and sign in form

## Todo
LOG-4506

## Testing

Test locally.

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/ControlledSignInMethodEditBox.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  isSignUpPasswordRequired,
+  isSignUpVerificationRequired,
+}: Props) => {
+  if (
+    !requiredSignInIdentifiers ||
+    isSignUpPasswordRequired === undefined ||
+    isSignUpVerificationRequired === undefined
+  ) {
+    return null;
+  }
```
Comment (charIeszhao) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/components/ControlledSignInMethodEditBox.tsx:28]:
```suggestion
  if (
    !requiredSignInIdentifiers ||
    !isSignUpPasswordRequired ||
    !isSignUpVerificationRequired
  ) {
    return null;
  }
```

File: packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/SignInForm.tsx [hunk 2]
```diff
@@ -12,22 +11,23 @@ import * as styles from './index.module.scss';
...
-  const setupVerificationAtSignUp = useWatch({
-    control,
-    name: 'signUp.verify',
-    defaultValue: false,
-  });
+  const { control, watch } = useFormContext<
+    Omit<SignInExperienceForm, 'signUp'> & {
+      signUp: Partial<SignInExperienceForm['signUp']>;
+    }
+  >();
```
Comment (charIeszhao) [inline packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/SignInForm.tsx:18]: Why not just modify the type in `SignInExperienceForm` directly?

Comment (xiaoyijun) [reply packages/console/src/pages/SignInExperience/tabs/SignUpAndSignInTab/SignInForm.tsx]: Updated

**Review by charIeszhao — APPROVED** [review]

**Comment by charIeszhao** [comment]
Don't forget to update PR title

**Pull request merged**